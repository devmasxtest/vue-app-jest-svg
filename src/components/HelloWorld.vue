<template>
  <div>
    <h1>{{message}}</h1>
    <figure>
      <img src="../assets/nodejs_logo.svg">
    </figure>
    <NodeLogo/>
  </div>
</template>
<script>
import NodeLogo from "./NodeLogo.vue";
import { setTimeout, setInterval } from "timers";
const genColor = () => {
  const colors = ["#233cccaa", "#cc239daa", "#b5971faa", "#b11515aa"];
  return colors[Math.floor(Math.random() * 3 + 0)];
};

export default {
  props: ["message"],
  components: {
    NodeLogo
  },
  mounted() {
    setInterval(() => {
      if (process.env.NODE_ENV != "test") {
        document.querySelector("#svg-js-logo").style.fill = genColor();
      }
    }, 1000);
  }
};
</script>
