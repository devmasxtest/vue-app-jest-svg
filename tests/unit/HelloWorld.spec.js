import { shallowMount } from '@vue/test-utils';
import HelloWorld from '@/components/HelloWorld.vue';

describe('HelloWorld', () => {
  it('renders props.msg when passed', () => {
    const wrapper = shallowMount(HelloWorld, { propsData: { message: 'Hello world!' }})
    expect(wrapper).toMatchSnapshot()
  })

})
