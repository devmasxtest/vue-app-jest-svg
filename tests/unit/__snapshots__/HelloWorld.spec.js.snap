// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld renders props.msg when passed 1`] = `
<div>
  <h1>Hello world!</h1>
  <nodelogosvg-stub></nodelogosvg-stub>
</div>
`;
